umbral/keys.py

```diff
@@ -10,13 +10,78 @@
 from cryptography.hazmat.backends.openssl.ec import (
     _EllipticCurvePublicKey, _EllipticCurvePrivateKey
 )
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
 
 from umbral.config import default_params
-from umbral.point import Point, BigNum
+from umbral.point import Point
+from umbral.bignum import BigNum, hash_to_bn
 from umbral.params import UmbralParameters
 
+from abc import ABC, abstractmethod
+ 
+class AbstractUmbralKeyingMaterial(ABC):
+
+    @abstractmethod
+    def _get_keying_material(self):
+        pass
+
+    def to_bytes(self, password: bytes=None, _scrypt_cost: int=20):
```

Comment:
Looks like you haven't merged the latest changes where I also pass in an encoder to allow arbitrary encoding of the bytes. See #112